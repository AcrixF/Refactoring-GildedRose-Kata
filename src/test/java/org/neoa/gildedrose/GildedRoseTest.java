package org.neoa.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class GildedRoseTest {

    @Test
    void foo() {
        CombinationApprovals.verifyAllCombinations(this::updateItem,
                new String []{"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
                new Integer []{-1, 0, 11, 15},
                new Integer []{0, 5, 49, 50});
    }

    private String updateItem(String name, int sellIn, int quality) {
        Item[] items = new Item[] {Item.createItem(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
