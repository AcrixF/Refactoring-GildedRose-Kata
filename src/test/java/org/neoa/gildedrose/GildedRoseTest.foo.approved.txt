[foo, -1, 0] => foo, -2, 0 
[foo, -1, 5] => foo, -2, 3 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 50] => foo, -2, 48 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 5] => foo, -1, 3 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 50] => foo, -1, 48 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 5] => foo, 10, 4 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 50] => foo, 10, 49 
[foo, 15, 0] => foo, 14, 0 
[foo, 15, 5] => foo, 14, 4 
[foo, 15, 49] => foo, 14, 48 
[foo, 15, 50] => foo, 14, 49 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 5] => Aged Brie, -2, 7 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 5] => Aged Brie, -1, 7 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 5] => Aged Brie, 10, 6 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 15, 0] => Aged Brie, 14, 1 
[Aged Brie, 15, 5] => Aged Brie, 14, 6 
[Aged Brie, 15, 49] => Aged Brie, 14, 50 
[Aged Brie, 15, 50] => Aged Brie, 14, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 5] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 5] => Backstage passes to a TAFKAL80ETC concert, 10, 6 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 15, 0] => Backstage passes to a TAFKAL80ETC concert, 14, 1 
[Backstage passes to a TAFKAL80ETC concert, 15, 5] => Backstage passes to a TAFKAL80ETC concert, 14, 6 
[Backstage passes to a TAFKAL80ETC concert, 15, 49] => Backstage passes to a TAFKAL80ETC concert, 14, 50 
[Backstage passes to a TAFKAL80ETC concert, 15, 50] => Backstage passes to a TAFKAL80ETC concert, 14, 50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 5] => Sulfuras, Hand of Ragnaros, -1, 5 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 5] => Sulfuras, Hand of Ragnaros, 11, 5 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 15, 0] => Sulfuras, Hand of Ragnaros, 15, 0 
[Sulfuras, Hand of Ragnaros, 15, 5] => Sulfuras, Hand of Ragnaros, 15, 5 
[Sulfuras, Hand of Ragnaros, 15, 49] => Sulfuras, Hand of Ragnaros, 15, 49 
[Sulfuras, Hand of Ragnaros, 15, 50] => Sulfuras, Hand of Ragnaros, 15, 50 
